django/template/context.py

```diff
@@ -251,7 +251,14 @@ def bind_template(self, template):
         processors = template.engine.template_context_processors + self._processors
         updates = {}
         for processor in processors:
-            updates.update(processor(self.request))
+            try:
+                updates.update(processor(self.request))
+            except TypeError as e:
+                raise TypeError(
+                    "Context processor %s didn't return a dictionary."
+                    % processor.__name__
```

Comment:
I'd use f-string and `__qualname__`:
```suggestion
                    f"Context processor {processor.__qualname__} didn't return a dictionary."
```

Reply:
Actually just noticed the extra `f` on this, have removed it.